sky/shell/platform/ios/sky_dynamic_service_loader.mm

```diff
@@ -25,6 +28,48 @@ - (SkyDynamicServiceHandler)serviceEntryPoint;
 
 @end
 
+enum class InstallSystemThunksResult {
+  Failure,
+  EmbedderOlder,
+  EmbedderNewer,
+  Success,
+};
+
+static InstallSystemThunksResult InstallSystemThunksInLibrary(
+    void* library_handle) {
+  if (library_handle == NULL) {
+    return InstallSystemThunksResult::Failure;
+  }
+
+  dlerror();
+  MojoSetSystemThunksFn set_thunks_fn = reinterpret_cast<MojoSetSystemThunksFn>(
+      dlsym(library_handle, kMojoSetSystemThunksFnName));
+
+  if (set_thunks_fn == NULL || dlerror() != NULL) {
+    return InstallSystemThunksResult::Failure;
+  }
+
+  MojoSystemThunks embedder_thunks = MojoMakeSystemThunks();
+
+  size_t result = set_thunks_fn(&embedder_thunks);
+
+  if (result > sizeof(MojoSystemThunks)) {
+    // The dylib expects to use a system thunks table that is larger than what
+    // is currently supported by the embedder. This indicates that the embedder
+    // is older than the dylib.
+    return InstallSystemThunksResult::EmbedderOlder;
+  }
+
+  if (result < sizeof(MojoSystemThunks)) {
+    // The dylib expect to use the system thunks table to be smaller than
+    // what is currently supported by the embedder. The exisiting entries in
+    // the table are stable.
+    return InstallSystemThunksResult::EmbedderNewer;
```

Comment:
not sure it's really useful to distinguish this from success. the thunk table is designed to support running libraries that link in older (smaller) versions of the table just fine